### PR TITLE
merge the null_safety branch into master

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

**Comment by natebosch** [comment]
@kevmoo - this should have been a squash commit... the history of `master` is really hard to follow now.

For future packages, we are using squash merges for the PR that applies `null_safety` changes onto `master`

**Comment by kevmoo** [comment]
> @kevmoo - this should have been a squash commit... the history of `master` is really hard to follow now.
> 
> For future packages, we are using squash merges for the PR that applies `null_safety` changes onto `master`

Sorry, @natebosch – understood!
